### PR TITLE
Update to Apache Jena 4.2

### PR DESCRIPTION
**Pull request by egonw**

### CODE REVIEW

**Comment by egonw** [comment]
@johnmay, one attention point is the support of Java8. Java11  was released in 2018 and is a LTS release, and last week Java17 was released the second LTS release after Java8. Dependencies are dropping Java8 support, maybe we should too?

File: storage/iordf/pom.xml [hunk 1]
```diff
@@ -21,19 +21,12 @@
...
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-log4j12</artifactId>
-            <version>1.7.10</version>
-            <scope>test</scope>
+            <version>4.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.jena</groupId>
+            <artifactId>jena-arq</artifactId>
+            <version>4.2.0</version>
```
Comment (javadev) [inline storage/iordf/pom.xml:29]: image

It looks like the latest version for java 1.8 is 3.9.0

**Comment by egonw** [comment]
@johnmay, is 2.8 going to be Java 11?

**Comment by johnmay** [comment]
We can actually make just this module Java 11 right now, since I don't use RDF/JENA it wouldn't get in the way.

**Comment by egonw** [comment]
@johnmay, ready to go into master.

**Comment by johnmay** [comment]
Can work out why the pipelines have not run, do we need Java 11?

**Comment by egonw** [comment]
Yes, sorry, we did not migrate in the next upcoming release yet? Sorry, my bad!

**Comment by johnmay** [comment]
@egonw This is good to merge then can do a release

**Comment by egonw** [comment]
> @egonw This is good to merge then can do a release

@johnmay, I double check, update to Jena 4.5 (minor update) and then merge in, okay?

**Comment by johnmay** [comment]
The minor update doesn't fix anything...? This is a major update.

**Comment by egonw** [comment]
> The minor update doesn't fix anything...? This is a major update.

This is the patch I have in mind, updating to the latest dependencies (to make it just a bit more future proof):

```
diff --git a/storage/iordf/pom.xml b/storage/iordf/pom.xml
index 3fac744a51..3f5b29e04d 100644
--- a/storage/iordf/pom.xml
+++ b/storage/iordf/pom.xml
@@ -17,7 +17,7 @@
         <dependency>
             <groupId>org.apache.jena</groupId>
             <artifactId>jena-core</artifactId>
-            <version>4.4.0</version>
+            <version>4.5.0</version>
             <exclusions>
                 <exclusion>
                     <artifactId>slf4j-log4j12</artifactId>
@@ -32,7 +32,7 @@
         <dependency>
             <groupId>org.apache.jena</groupId>
             <artifactId>jena-arq</artifactId>
-            <version>4.4.0</version>
+            <version>4.5.0</version>
             <exclusions>
                 <exclusion>
                     <artifactId>slf4j-log4j12</artifactId>
@@ -47,7 +47,7 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-log4j12</artifactId>
-            <version>1.7.10</version>
+            <version>1.7.36</version>
             <exclusions>
                 <exclusion>
                     <groupId>log4j</groupId>
```

**Comment by egonw** [comment]
@johnmay, this PR: https://github.com/cdk/cdk/pull/867

**Comment by johnmay** [comment]
Sorry not clear, make it 4.5.0 you mean?

**Comment by egonw** [comment]
Yes, 4.5 was released very recently as an update to 4.2/4.4. See the PR.

**Comment by johnmay** [comment]
Fixed

**Comment by egonw** [comment]
Do you merge this in, or should I?

**Pull request merged**